libraries/cmd/unit_generic.cpp

```diff
@@ -1397,11 +1397,21 @@ void Unit::ProcessDeleteQueue() {
 #ifdef DESTRUCTDEBUG
         if ( unit_delete_queue.back()->isSubUnit() ) {
             VS_LOG(debug, "Subunit Deleting (related to double dipping)");
+            unit_delete_queue.pop_back();
+            continue;
```

Comment:
This code only comes into effect when `DESTRUCTDEBUG` is defined. So not very often.